web_model_read_only/tests/test_models.py

```diff
@@ -0,0 +1,82 @@
+from odoo.exceptions import AccessError
+from odoo.tests.common import SavepointCase
+
+
+class TestIrModel(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestIrModel, cls).setUpClass()
+        cls.readonly_group = cls.env.ref("base.group_user")
+        cls.group_manager = cls.env.ref("base.group_erp_manager")
+        cls.ir_model = cls.env["ir.model"].create(
+            {"name": "x_test_model", "model": "x_test.model"}
+        )
+        User = cls.env["res.users"]
+        cls.manager = User.create(
+            {
+                "name": "Manager",
+                "login": "Manager",
+                "groups_id": cls.group_manager,
+            }
+        )
+        cls.user_readonly = User.create(
+            {
+                "name": "User readonly",
+                "login": "user_readonly",
+                "groups_id": cls.readonly_group,
+            }
+        )
+
+    def test_check_access_rights(self):
```

Comment:
Please split this function on the separate cases.
And will make more tests for the coverage all cases.